### PR TITLE
Rem 623

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
a new page properties , id , name in the form , resource combobox in the index for changing the fetch of the data

### CODE REVIEW

File: src/pages/iv-properties/forms/PropertiesForm.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      } else toast.success(platformLabels.Edited)
+
+      invalidate()
+      window.close()
+    }
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
```
Comment (imtithalhalalo) [inline src/pages/iv-properties/forms/PropertiesForm.js:56]: remove try/catch

File: src/pages/iv-properties/forms/PropertiesForm.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        if (id) {
+          const res = await getRequest({
+            extension: InventoryRepository.Dimension.get,
+            parameters: `_dimension=${dimNum}&_id=${id}`
+          })
+
+          if (res && res.record) {
```
Comment (imtithalhalalo) [inline src/pages/iv-properties/forms/PropertiesForm.js:63]: why do you need this if?

File: src/pages/iv-properties/index.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    } catch (error) {}
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [formik.values.dimValue])
+
+  useEffect(() => {
+    ;(async function () {
+      try {
```
Comment (imtithalhalalo) [inline src/pages/iv-properties/index.js:124]: remove try/catch

File: src/pages/iv-properties/index.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+              filter={item => item.value !== ''}
+              error={!formik.values.dimValue}
+            />
+          </Grid>
+        </Grid>
+      </Fixed>
+      <Grow>
+        <Table
+          columns={rowColumns}
+          gridData={data || []}
```
Comment (imtithalhalalo) [inline src/pages/iv-properties/index.js:166]: do you need empty array?

**Comment by imtithalhalalo** [comment]
when I added a color then clicked save it showed empty records but when I reload it show them normally
Screenshot 2024-10-16 132400

File: src/pages/iv-properties/forms/PropertiesForm.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      name: yup.string().required(),
+      id: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      const id = obj?.id
```
Comment (imtithalhalalo) [inline src/pages/iv-properties/forms/PropertiesForm.js:39]: use obj?.id directly no need for constant

File: src/pages/iv-properties/forms/PropertiesForm.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+            formik.setValues(res.record)
+          }
+        }
+      } catch (exception) {}
+    })()
+  }, [id])
+
+  return (
+    <FormShell
+      resourceId={ResourceIds.IVDimension}
```
Comment (imtithalhalalo) [inline src/pages/iv-properties/forms/PropertiesForm.js:73]: is the resource id already added?

**Comment by FatimaSalami** [comment]
image

in asp we can't clear the combo
it is always mandatory to keep one of selections
so it can't be deleted, it can be replaced

and can we have some distance between the combo and the grid upper margin??
its shape will look better

**Review by FatimaSalami — APPROVED** [review]

**Pull request merged**